crates/circuit/src/symbol_expr.rs

```diff
@@ -2526,7 +2621,10 @@ impl PartialOrd for SymbolExpr {
 
 impl From<&str> for SymbolExpr {
     fn from(v: &str) -> Self {
-        SymbolExpr::Symbol(Box::new(v.to_string()))
+        SymbolExpr::Symbol {
```

Comment:
It might be better to implement `From<String>` instead and move the string into your `Box`.